improver_tests/fire_weather/test_build_up_index.py

```diff
@@ -0,0 +1,366 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+import warnings
+
+import numpy as np
+import pytest
+from iris.cube import Cube, CubeList
+
+from improver.fire_weather.build_up_index import BuildUpIndex
+from improver_tests.fire_weather import make_cube, make_input_cubes
+
+
+def input_cubes(
+    dmc_val: float = 10.0,
+    dc_val: float = 15.0,
+    shape: tuple[int, int] = (5, 5),
+    dmc_units: str = "1",
+    dc_units: str = "1",
+) -> list[Cube]:
+    """Create a list of dummy input cubes for BUI tests, with configurable units.
+
+    DMC cube has time coordinates; DC cube does not (following the pattern).
+
+    Args:
+        dmc_val (float): DMC value for all grid points.
+        dc_val (float): DC value for all grid points.
+        shape (tuple[int, int]): Shape of the grid for each cube.
+        dmc_units (str): Units for DMC cube.
+        dc_units (str): Units for DC cube.
+
+    Returns:
+        list[Cube]: List of Iris Cubes for DMC and DC.
+    """
+    return make_input_cubes(
+        [
+            ("duff_moisture_code", dmc_val, dmc_units, True),
+            ("drought_code", dc_val, dc_units, False),
+        ],
+        shape=shape,
+    )
+
+
+@pytest.mark.parametrize(
+    "dmc_val, dc_val, expected_bui",
+    [
+        # Case 0: Both zero (special case)
+        (0.0, 0.0, 0.0),
+        # Case 1: DC only (DMC=0) -> BUI should be 0
+        (0.0, 50.0, 0.0),
+        # Case 2: Small DMC, large DC (DMC <= 0.4*DC)
+        (5.0, 30.0, 7.06),
+        # Case 3: DMC=10, DC=30 (DMC <= 0.4*DC)
+        (10.0, 30.0, 10.91),
+        # Case 4: DMC > 0.4*DC case
+        (20.0, 30.0, 19.76),
+        # Case 5: Equal DMC and DC (DMC > 0.4*DC)
+        (50.0, 50.0, 49.41),
+        # Case 6: High DC (DMC <= 0.4*DC)
+        (30.0, 200.0, 43.64),
+        # Case 7: High DMC and DC (DMC > 0.4*DC)
+        (45.9, 123.9, 47.67),
+        # Case 8: High values with DMC > 0.4*DC
+        (100.0, 150.0, 99.46),
+    ],
+)
+def test__calculate(
+    dmc_val: float,
+    dc_val: float,
+    expected_bui: float,
+) -> None:
+    """Test calculation of BUI from DMC and DC.
+
+    Verifies BUI calculation from DMC and DC values.
+
+    Args:
+        dmc_val (float): DMC value to test.
+        dc_val (float): DC value to test.
+        expected_bui (float): Expected BUI value.
+    """
+    cubes = input_cubes(dmc_val=dmc_val, dc_val=dc_val)
+    plugin = BuildUpIndex()
+    plugin.load_input_cubes(CubeList(cubes))
```

Comment:
With the suggested pytest fixture, this would become
```suggestion
def test__calculate(
    dmc_val: float,
    dc_val: float,
    input_cubes,
    expected_bui: float,
) -> None:
    """Test calculation of BUI from DMC and DC.

    Verifies BUI calculation from DMC and DC values.

    Args:
        dmc_val (float): DMC value to test.
        dc_val (float): DC value to test.
        expected_bui (float): Expected BUI value.
    """
    plugin = BuildUpIndex()
    plugin.load_input_cubes(input_cubes)
```

Similar changes would be needed to all the other tests too, and while this might be more pythonic, and also better aligned with other IMPROVER tests, it isn't compulsory.